app/js/authentication_controller.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+/* global User, Utils */
+
+(function(exports) {
+  function update_error_list(list, errors) {
+    list.innerHTML = '';
+    if (errors) {
+      for (var i in errors) {
+        var error = document.createElement('li');
+        error.appendChild(document.createTextNode(errors[i]));
+        list.appendChild(error);
+      }
+    }
+  }
+
+  function authenticate(evt) {
+    evt.preventDefault();
+
+    var form = document.getElementById('login');
+    var username = form.elements.username.value;
+    var password = form.elements.password.value;
+
+    var promise = User.authenticate_user(username, password);
+
+    Utils.toggle_spinner();
+
+    promise.then(function(token) {
+      User.set_user(username, password, token).then(function() {
+        Utils.toggle_spinner();
+      });
+    }).catch(function(response) {
+      var errors = JSON.parse(response);
+
+      update_error_list(document.getElementById('login_errors'),
+        errors.non_field_errors);
+
+      update_error_list(document.getElementById('username_errors'),
+        errors.username);
+
+      update_error_list(document.getElementById('password_errors'),
+        errors.password);
+
+      Utils.toggle_spinner();
+    });
+  }
+
+  function request_password_reset(evt) {
```

Comment:
I'd prefer to keep controllers simple. We're being bitten by questions_controller.js having too much logic. So a password_reset_controller.js makes sense to me. And you can move update_error_list to utils.js ? (although I hate files that are just a grab bag of functions)